nixos/modules/services/hardware/amdvlk.nix

```
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.hardware.amdgpu.amdvlk;
+in {
+  options.hardware.amdgpu.amdvlk = {
+    enable = lib.mkEnableOption "AMDVLK Vulkan driver";
+
+    package = lib.mkPackageOption pkgs "amdvlk" { };
+
+    supportExperimental.enable = lib.mkEnableOption "Experimental features support";
+
+    support32Bit.enable = lib.mkEnableOption "32-bit driver support";
+    support32Bit.package = lib.mkPackageOption pkgs.driversi686Linux "amdvlk" { };
+
+    settings = lib.mkOption {
+      type = lib.types.attrs;
+      default = { };
+      example = {
+        AllowVkPipelineCachingToDisk = 1;
+        ShaderCacheMode = 1;
+        IFH = 0;
+        EnableVmAlwaysValid = 1;
+        IdleAfterSubmitGpuMask = 1;
+      };
+      description = ''
+        Runtime settings for AMDVLK to be configured {file}`/etc/amd/amdVulkanSettings.cfg`.
+        See [AMDVLK GitHub page](https://github.com/GPUOpen-Drivers/AMDVLK?tab=readme-ov-file#runtime-settings).
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    hardware.opengl = lib.mkMerge [
+      {
+        enable = lib.mkDefault true;
+        driSupport = lib.mkDefault true;
+        extraPackages = [ cfg.package ];
+      }
+      (lib.mkIf cfg.support32Bit.enable {
+        driSupport32Bit = lib.mkDefault true;
+        extraPackages32 = [ cfg.support32Bit.package ];
+      })
+    ];
```

```suggestion
    hardware.opengl = {
      enable = true;
      driSupport = true;
      extraPackages = [ cfg.package ];
      driSupport32Bit = lib.mkIf cfg.support32Bit.enable true;
      extraPackages32 = [ cfg.support32Bit.package ];
    }
```
extraPackages32 is completely unused, if driSupport32Bit is false.
Also we don't need to use mkDefault because we want to enable those options if the module is enabled and people should know, if they are disabled

---

Done in https://github.com/NixOS/nixpkgs/pull/318175/commits/52e48f6f78cd1c3c5216caac8e58471b2303f213